python/pyspark/__init__.pyi

```diff
@@ -0,0 +1,73 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import Callable, Optional, TypeVar
+
+from pyspark.accumulators import (  # noqa: F401
+    Accumulator as Accumulator,
+    AccumulatorParam as AccumulatorParam,
```

Comment:
quick question, why should we do `as` to the same import?

Reply:
In case of stub packages this allows us to skip annotations for certain modules ‒ if stub file is not found, type checker will go and inspect the module itself.

In theory, if stub files reside within the same directory structure as the source, it shouldn't be necessary.  But since things are not always smooth, I'd prefer to keep this.

Reply:
With MyPy can disable implicit reexport ‒ that's default configuration for typeshed.

If we replaced above with

```python
from pyspark.accumulators import  Accumulator
```

we couldn't do 

```python
from pyspark import Accumulator
```

Technically speaking we could also provide `__all__` in the stubs ‒ I just wasn't aware of that back when I started working on stubs.